docs/develop/howtos/search_terms_migration.md

```diff
@@ -0,0 +1,39 @@
+# Create search terms mappings
+
+This guide describes the process to create search terms mappings in the application layer.
+
+This is particularly useful when a certain search term is not supported anymore by the search engine, in favour of a new term, e.g., when a field's name is modified in a mapping.
+
+The query is parsed in the application layer and the old term is replaced with the new one before passing the query to the search engine.
+
+## Steps
+
+- Identify the legacy search term (e.g. `q=old_term:"value"`).
```

Comment:
I wouldn't restrict this to `legacy` use cases as we can use the mapping just to strip some nested paths like `"title" : "metadata.title"`